### PR TITLE
Tabular: Fixed balanced_accuracy scoring issue with missing classes

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Changes internal learner scorer to use external label values in eval_metric calls rather than internal label values. This fixes edge case issues in `balanced_accuracy` when multiple new test time classes are provided and are all converted to class `-1` internally, causing an incorrect score to be returned.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tabular/src/autogluon/tabular/learner/abstract_learner.py [hunk 1]
```diff
@@ -170,30 +170,26 @@ def transform_features(self, X):
...
         X = self.transform_features(X)
-        y = self.label_cleaner.transform(y)
-        trainer = self.load_trainer()
-        if self.problem_type == MULTICLASS:
-            y = y.fillna(-1)
-            if (not trainer.eval_metric_expects_y_pred) and (-1 in y.unique()):
-                # log_loss / pac_score
-                raise ValueError(f'Multiclass scoring with eval_metric=\'{self.eval_metric.name}\' does not support unknown classes.')
+        y_internal = self.label_cleaner.transform(y)
+        y_internal = y_internal.fillna(-1)
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/learner/abstract_learner.py:190]: doesn't balanced_accuracy bug still exist in score_debug() if two different unknown classes are replaced by -1?

Comment (Innixma) [reply tabular/src/autogluon/tabular/learner/abstract_learner.py:191]: No, because y_internal is only used in pred_proba metrics, but balanced_accuracy is a pred metric.

**Comment by szha** [comment]
Job PR-851-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-851/2/index.html

**Review by jwmueller — APPROVED** [review]
looks good

**Comment by Innixma** [comment]
Benchmarking shows no errors or accuracy drop.
